framework/wazuh/tests/test_decoder.py

```diff
@@ -257,25 +269,33 @@ def test_upload_file_ko(mock_remote_commands, mock_safe_move, mock_xml, mock_del
     assert result.render()['data']['failed_items'][0]['error']['code'] == 1112, 'Error code not expected.'
 
     # Error doing backup
-    with patch('wazuh.decoder.exists'):
-        result = decoder.upload_decoder_file(filename='test_decoders.xml', content='test', overwrite=True)
-        assert isinstance(result, AffectedItemsWazuhResult), 'No expected result type'
-        assert result.render()['data']['failed_items'][0]['error']['code'] == 1019, 'Error code not expected.'
+    result = decoder.upload_decoder_file(filename='test3_decoders.xml', content='test', overwrite=True)
+    assert isinstance(result, AffectedItemsWazuhResult), 'No expected result type'
+    assert result.render()['data']['failed_items'][0]['error']['code'] == 1019, 'Error code not expected.'
 
 
-def test_delete_decoder_file():
+@pytest.mark.parametrize('filename, relative_dirname', [
+    ('test1_decoders.xml', None),
+    ('test1_decoders.xml', None),
+    ('test3_decoders.xml', None),
+    ('test2_decoders.xml', 'tests/data/etc/decoders/subpath'),
+    ('test3_decoders.xml', 'tests/data/etc/decoders/subpath'),
+    ('test3_decoders.xml', 'tests/data/etc/decoders/subpath/'),
+])

```

Comment:
```suggestion
@pytest.mark.parametrize('filename, relative_dirname', [
    ('test1_decoders.xml', None),
    ('test2_decoders.xml', None),
    ('test3_decoders.xml', None),
    ('test1_decoders.xml', 'tests/data/etc/decoders/subpath'),
    ('test2_decoders.xml', 'tests/data/etc/decoders/subpath'),
    ('test3_decoders.xml', 'tests/data/etc/decoders/subpath'),
])

```

Reply:
I Agree with the first comment.
Regarding the second comment, it uses different relative_dirname paths, with and without trailing slash.